### ISSUE TITLE
Insufficient memory for daemonset - no new node

### ISSUE THREAD

**Issue by JulesClaussen** [aws/karpenter-provider-aws#7406]
### Description

**Observed Behavior**:

Sometimes, (I can't understand when) we have some daemonsets pods that cannot be scheduled, due to "Insufficient Memory".
Karpenter doesn't start a new (bigger) node that would allow the daemon set to be scheduled.
This is not the same issue as https://github.com/kubernetes-sigs/karpenter/issues/731. This occurs for older daemonsets as well, and for nodes created recently

**Expected Behavior**:
Karpenter should start a new node that would allow all daemonsets and workload pods to be scheduled.

**Reproduction Steps** (Please include YAML):
I don't know how to reproduce. This happens often, but I can't pinpoint why it does, and why it doesn't.
A node in question contains some kube-system pods (aws-node, kube-proxy, secrets-store) but also some workload pods (that have PDB)

**Versions**:
- Chart Version: 1.0.8
- Kubernetes Version (`kubectl version`): 1.31.0

**Comment by mariuskimmina** [comment]
Just a daemonset? Karpenter will not create a new node for just a daemonset, that would be the same as an empty node. Also, I believe daemonsets should be first on the node before any workload, so having insufficient memory for that sounds very odd - would be great if you could capture some logs and events the next time this happens.

**Comment by JulesClaussen** [comment]
I agree that a daemonset on it's own is pointless. But the daemonset here is used to collect some instance network metrics (it's ethtool-exporter). 

Not sure what kinda logs you would like to see. Those are my failed pods events:
```
  Warning  FailedScheduling  57m (x161 over 85m)    default-scheduler  0/23 nodes are available: 1 Insufficient memory. preemption: 0/23 nodes are available: 1 No preemption victims found for incoming pod, 22 Preemption is not helpful for scheduling.
  Warning  FailedScheduling  2m54s (x315 over 86m)  default-scheduler  0/22 nodes are available: 1 Insufficient memory. preemption: 0/22 nodes are available: 1 No preemption victims found for incoming pod, 21 Preemption is not helpful for scheduling.
```

Karpenter does not show anything. Those events might not be picked up by Karpenter, so I don't have much logs from karpenter side unfortunately.
I have looked for Karpenter logs over the last days, and I seem to have logs only about registered and initialized. 
Should I look at something more specific @mariuskimmina ?

**Comment by mariuskimmina** [comment]
You could try to add `priorityClassName: system-node-critical` to your daemonset, this way it should evict other workloads and make room for daemonset. The workload being unschedulable would then trigger Karpenter to create new nodes (where as a dangling daemonset does not). 

See also https://github.com/kubernetes-sigs/karpenter/issues/731

**Comment by JulesClaussen** [comment]
Yes, this is indeed a good workaround that we implemented for other daemonsets as they're more critical. But shouldn't Karpenter be able to handle that? I'm surprised there are no logs in karpenter, so it looks like the event is not processed (or it is but nor verbose)

**Comment by ebizboy** [comment]
I also encountered an issue where pods couldn't be scheduled on nodes after assigning resources to daemonsets. To avoid this problem, I didn't assign resources to the daemonsets and instead allocated slightly more generous resources to deployments and statefulsets. It would be great if Karpenter could handle this.

**Comment by pznamensky** [comment]
It's exactly the same for us. Can we somehow help pinpoint the problem?

**Comment by ebizboy** [comment]
@mariuskimmina 

"priorityClassName: system-node-critical" solved my problem. Thank you

**Comment by gnuletik** [comment]
On our side, we added tolerations on our Daemonsets to make our daemonsets start before other workloads.

**Comment by JulesClaussen** [comment]
@rschalo out of curiosity, do you have any leads on this one? Do you believe there's a mistake in the calculation at some point? Do you take into account when there are multiple containers in the same DS pod?
